var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -33,6 +33,9 @@ class Hdf5(CMakePackage):
 
     license("custom")
 
+    depends_on("cxx", type="build", when="+cxx")
+    depends_on("fortran", type="build", when="+fortran")
```

Comment:
@alalazo Why do these have to be build dependencies? I think a default deptype is fine here, as this will eventually resolve to a build dependency on a compiler and link dependencies on libraries.

Reply:
Discussed in DMs. We might want to consider a new deptype. Currently the dependency maps to the compiler only, which needs to be `build` due to unification rules (once compilers are real dependencies).